### DISCUSSION TITLE
Visualize annotation from JSON for further modification

### DISCUSSION THREAD

**Discussion by umayerr** [tecoholic/ner-annotator#98] [Ideas]
Hi @alvi-khan, I must admit this is a super annotation tool. Thank you for the efforts. However, I am wondering if there is a feature of uploading the JSON (previously exported) file for further correction in annotated data. 
At the current state it allows text to annotate with an wonderful color coding. But if we need to make some correction in the annotation file which is typically in JSON format then it needs to be re-annotated from scratch. So, the tool might have an option of uploading the JSON which will render the same color coded visualization that it had while doing annotation. Then further correction (removing wrongly annotated tokens and adding new tokens in annotation) would be easier. Finally, the corrected file will also be downloaded in JSON format.

**Comment by alvi-khan** [comment]
Hello @umayerr. Is it possible that you are using an older version of the app? The current version supports importing the annotation files (from the menu `Annotations` → `Import`). I am attaching a screenshot below. This is available both in the last desktop versions ([here](https://github.com/tecoholic/ner-annotator/releases/latest)) and the web version ([here](https://tecoholic.github.io/ner-annotator/)).

![image](https://github.com/tecoholic/ner-annotator/assets/55395950/c2f065cc-e207-47c5-9338-e015687f61fe)

**Reply by umayerr** [reply]
Hi, sorry for the late reply. 
Thanks for your information, I was not aware about that update. I have another observation/question though on that as I am actively using this tool for my project.
1. I cannot go to the page (you attached as screenshot) directly. We need to upload a text file to get into that page. Do I need to upload original corresponding text file before importing annotation file? If so, it seems like I cannot open an annotation if I do not have the original text file.
2. I went to the page (you attached as screenshot) by uploading a single line text file. Then I uploaded the annotation which was different from that uploaded text file. It did not work. It tried to annotated the single text file using the labels from the annotation file.
Thank you!

**Reply by alvi-khan** [reply]
Hello @umayerr! Yes, you will need to upload the original corresponding text file before importing the annotation file.

**Reply by umayerr** [reply]
Thank you so much @alvi-khan for all of your clarifications.
